### ISSUE TITLE
PostgresSQL 14 DBs are requesting a  root password 

### ISSUE THREAD

**Issue by kev-pinto-cts** [terraform-google-modules/terraform-google-sql-db#373]
### TL;DR

The Module works fine with Postgres database Version 10, however when one uses Version 14(>=11), it throws an error stating 
Error: Error, failed to create instance pginst: googleapi: Error 400: Missing parameter: Root Password., required

The TF page has the following note which seems related to the issue
NOTE on google_sql_database_instance:
- Second-generation instances include a default 'root'@'%' user with no password. This user will be deleted by Terraform on instance creation. You should use google_sql_user to define a custom user with a restricted host and strong password.

### Expected behavior

It should have created the Instance with PostgresSQL 14

### Observed behavior

module.sql-db_postgresql["pginst"].google_sql_database_instance.default: Creating...
╷
│ Error: Error, failed to create instance pginst: googleapi: Error 400: Missing parameter: Root Password., required
│ 
│   with module.sql-db_postgresql["pginst"].google_sql_database_instance.default,
│   on .terraform/modules/sql-db_postgresql/modules/postgresql/main.tf line 48, in resource "google_sql_database_instance" "default":
│   48: resource "google_sql_database_instance" "default" {

### Terraform Configuration

```hcl
# backend.tf
terraform {
  backend "gcs" {
    bucket = "XXX-99999-tfstate"
    prefix = "cloudsql"
  }
}

#provider.tf
terraform {
  required_providers {
    google = {
      source  = "hashicorp/google"
      version = "4.28.0"
    }
  }
}

provider "google" {
  project = "XXX"
  region  = "europe-west2"
}

#variables.tf
variable "location" {}

#terraform.tfvars
location = "europe-west2"

#main.tf
locals {
  config                   = yamldecode(file("./config/postgres.yaml"))
  default_user_name        = "postgres"
  default_database_version = "14"
  default_user_labels      = {
    project = data.google_project.nwmmr.project_id
    region  = var.location
  }
}

data "google_project" "nwmmr" {
}

module "sql-db_postgresql" {
  source   = "GoogleCloudPlatform/sql-db/google//modules/postgresql"
  version  = "13.0.0"
  for_each = {for instance in local.config.instances : instance.name => instance}

  name                 = each.value.name
  project_id           = data.google_project.nwmmr.project_id
  region               = try(each.value.region, var.location)
  zone                 = "${var.location}-a"
  deletion_protection  = false
  availability_type    = try(each.value.availability_type, "ZONAL")
  database_version     = try(each.value.database_version, local.default_database_version)
  activation_policy    = try(each.value.activation_policy, "ALWAYS")
  additional_databases = try(each.value.databases, [])
  enable_default_db    = length(try(each.value.databases, [])) > 0 ? false : true
  user_labels          = merge(try(each.value.user_labels, {}), local.default_user_labels)
}

#postgres.yaml
instances:
  - name: pginst
    availability_type: REGIONAL
    database_flags:
    - name: cloudsql.logical.decoding
      value: On
    databases:
    - name: db1
      charset: UTF8
      collation: en_US.UTF8
    - name: db2
      charset: UTF8
      collation: en_US.UTF8
    - name: db3
      charset: UTF8
      collation: en_US.UTF8
```

### Terraform Version

```sh
Terraform v1.3.3
on darwin_arm64
```

**Comment by arif17ginanjar** [comment]
Same issue here :(
🙏
